imageroot/actions/configure-module/20config

```diff
@@ -0,0 +1,91 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see COPYING.
+#
+
+#
+# Create a virtual host configuration
+# Input example:
+#  {"instance": "module1", "url": "http://127.0.0.0:2000", "path": "/foo", "http2https": true}
+#
+
+import json
+import sys
+import os
+import agent
+import tempfile
+
+# Try to parse the stdin as JSON.
+# If parsing fails, output everything to stderr
+data = json.load(sys.stdin)
+
+agent_id = os.getenv("AGENT_ID", "")
+if not agent_id:
+    raise Exception("AGENT_ID not found inside the environemnt")
+
+# Connect to redis
+r = agent.redis_connect(privileged=True).pipeline()
+
+restart_webapp = False
+
+webtop_request_https_certificate = os.environ["WEBTOP_REQUEST_HTTPS_CERTIFICATE"].lower() in ('true', '1', 't')
+if data.get("request_https_certificate") is not None:
+    if data.get("request_https_certificate") != webtop_request_https_certificate:
+        webtop_request_https_certificate = data["request_https_certificate"]
+        agent.set_env("WEBTOP_REQUEST_HTTPS_CERTIFICATE", data["request_https_certificate"])
+
+# Configure Traefik to route WebTop's host requests to the webtop module
+response = agent.tasks.run(
+    agent_id=agent.resolve_agent_id('traefik@node'),
+    action='set-route',
+    data={
+        'instance': os.environ['MODULE_ID'],
+        'url':  'http://127.0.0.1:' + os.environ["TCP_PORT"],
+        'http2https': True,
+        'lets_encrypt': webtop_request_https_certificate,
+        'host': data["hostname"],
+    },
+)
+# Check if traefik configuration has been successfull
+agent.assert_exp(response['exit_code'] == 0)
+
+if data["hostname"] != os.getenv("WEBTOP_HOSTNAME"):
+
+        public_url = 'https://' + data["hostname"] + '/webtop'
+        dav_url = 'https://' + data["hostname"] + '/webtop-dav/server.php'
+
+        tmp_sql_file = tempfile.NamedTemporaryFile()
+
+        tmp_sql_file.write(b"DELETE FROM \"core\".\"settings\" WHERE service_id = 'com.sonicle.webtop.core' and key = 'public.url';\n")
+        tmp_sql_file.write(b"INSERT INTO \"core\".\"settings\" (\"service_id\", \"key\", \"value\") VALUES ('com.sonicle.webtop.core', 'public.url', \'" + public_url.encode() + b"\');\n")
+
+        tmp_sql_file.write(b"DELETE FROM \"core\".\"settings\" WHERE service_id = 'com.sonicle.webtop.core' and key = 'davserver.url';\n")
+        tmp_sql_file.write(b"INSERT INTO \"core\".\"settings\" (\"service_id\", \"key\", \"value\") VALUES ('com.sonicle.webtop.core', 'davserver.url',\'" + dav_url.encode() + b"\');\n")
+
+        tmp_sql_file.flush()
+
+        os.system(f"podman cp {tmp_sql_file.name} postgres:/tmp/settings.sql")
+        os.system('podman exec postgres sh -c "psql -U postgres webtop5 < /tmp/settings.sql > /dev/null"')
```

Comment:
:thought_balloon: As alternative to creating/copying the tempfile it is possible to fork `psql` and send SQL commands to it through a pipe. You can find examples that _reads_ from a subprocess: https://github.com/NethServer/ns8-core/search?q=subprocess.popen

See Python subprocess.Popen docs for more examples. It can handle encoding as well! Roughly it could become like:

```python
with subprocess.Popen(['podman', 'exec', '-i', 'postgres', 'psql', '-U', 'postgres', 'webtop5'], stdin=subprocess.PIPE, text=True) as psql:
    print("DELETE FROM...", file=psql)
    # ...more print()s as needed
agent.assert_exp(psql.returncode == 0) # check the command is succesfull
```

Reply:
Done :+1: